### PR TITLE
Added Calendar Assignment

### PR DESCRIPTION
**Pull request by MishraSagar**
Please review my Calendar Assignment. @anjanaNewput @Shikha-Shakarwar @mayur-vaidya

### CODE REVIEW

File: Calendar/index.html [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+
+    </head>
+    <body>
+        <div class="main">
+
+            <div class="month-year-container">
+                <i class="fa fa-chevron-left" aria-hidden="true" id="prev-click"></i>
+                <div class="month middle-content" id="month-name">
+                    <select name="month" id="select-month">
+                        <option value="0">JANUARY</option>
```
Comment (mayur-vaidya) [inline Calendar/index.html]: We can add the options using loop.

File: Calendar/index.html [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+                        <option value="11">DECEMBER</option>
+                    </select>
+                </div>
+                <div class="year middle-content" id="year-name">
+                    <select name="year" id="select-year"></select>
+                </div>
+                <i class="fa fa-chevron-right" aria-hidden="true" id="next-click"></i>
+            </div>
+
+            <div class="week-days">
```
Comment (mayur-vaidya) [inline Calendar/index.html:27]: Again: Implement this using loop.

File: Calendar/js/app.js [hunk 3]
```diff
@@ -0,0 +1,205 @@
...
+
+    generateBirthdays(lastDateOfCurrentMonth);
+
+    if ((year == date.getFullYear()) && (month == date.getMonth())) {
+        showCurrentDate();
+    }
+}
+
+ function showCurrentDate() {
+     var currentDateElement = document.querySelector('[data-month-date="' + (month + 1) + '-' + date.getDate() + '"]');
```
Comment (mayur-vaidya) [inline Calendar/js/app.js]: Instead of doing this... add/remove class from the element to change CSS

File: Calendar/js/app.js [hunk 4]
```diff
@@ -0,0 +1,205 @@
...
+    var currentDateElement;
+    var birthdayElement;
+    var birthdayBlockInnerHTML;
+    console.log('check');
+
+    for (var i = 1; i <= lastDate; i++) {
+        birthdaysOnDate = birthdays[String(month + 1)][String(i)];
+        console.log(birthdaysOnDate);
+        birthdayBlockInnerHTML = '';
+        if (birthdaysOnDate != undefined) {
```
Comment (mayur-vaidya) [inline Calendar/js/app.js]: Do we need this condition?

File: Calendar/js/app.js [hunk 5]
```diff
@@ -0,0 +1,205 @@
...
+            currentDateElement.innerHTML = '<div id="bb" class="birthdayBlock" onmouseover="onBirthdayHover()">' + birthdayBlockInnerHTML + '</div>';
+            currentDateElement.innerHTML += i + ' <i class="fa fa-birthday-cake"></i>';
+            currentDateElement.style.color = "#F96120" //"#A5DE60"; //"#D01315"
+        }
+    }
+}
+
+function onBirthdayHover() {
+        window.setInterval(function() {
+            var elem = document.getElementById('bb');
```
Comment (mayur-vaidya) [inline Calendar/js/app.js]: What is 'bb' here? Keep more meaningful ID name.

File: Calendar/js/app.js [hunk 6]
```diff
@@ -0,0 +1,233 @@
...
+    },
+    "12" : {
+        "3" : ["Abhijeet Bhowmik"],
+        "7" : ["Gourav Agrawal"],
+        "11" : ["Shashank Saxena"],
+        "12" : ["Nitesh Thakur", "Satya Narayan patidar"]
+    }
+}
+
+function createCalendar(_date, _month, _year) {
```
Comment (shikhashakarwar) [inline Calendar/js/app.js]: Do wee need to pass these in function parms`_date`, `_month`, `_year`  ?  As these are globally set.

**Comment by MishraSagar** [comment]
> Do we need to pass in function params `_date`, `_month`, `_year`? as these are globally set.

No mam. When I started the project, I kept them local. but I will modify the calls and push again.

File: Calendar/js/app.js [hunk 7]
```diff
@@ -0,0 +1,233 @@
...
+    var startNumber = 1;
+    var lastDateOfPreviousMonth;
+    var totalRows;
+    var i = 1;
+    var j = 1;
+    var k = 1;
```
Comment (shikhashakarwar) [inline Calendar/js/app.js]: Why declare `i`, `j` and `K`? Don't think you are using.

File: Calendar/js/app.js [hunk 8]
```diff
@@ -0,0 +1,233 @@
...
+
+    totalRows = Math.floor(((dayOnFirstOfMonth + lastDateOfCurrentMonth - 1) / 7) + 1);
+    weekOnLast = weekOnFirst + totalRows - 1;
+    dayOnFirstSaturday = 7 - dayOnFirstOfMonth;
+
+    for (var i = startNumber; i <= lastDateOfPreviousMonth; i++) {
+        _calendar.innerHTML += '<div class="dates prev-next-date"><span>' + i + '</span><div>';
+    }
+
+    for (var i = 1; i <= lastDateOfCurrentMonth; i++) {
```
Comment (shikhashakarwar) [inline Calendar/js/app.js:96]: You can handle current date highlighting in this loop. so no need of `showCurrentDate()`

File: Calendar/js/app.js [hunk 9]
```diff
@@ -0,0 +1,233 @@
...
+        }
+        else {
+            _weekElement.innerHTML += '<div class="week-block"></div>';
+        }
+    }
+
+    generateBirthdays(lastDateOfCurrentMonth);
+
+    if ((_year == _date.getFullYear()) && (_month == _date.getMonth())) {
+        showCurrentDate();
```
Comment (shikhashakarwar) [inline Calendar/js/app.js]: read above comment and remove this.

File: Calendar/js/app.js [hunk 10]
```diff
@@ -0,0 +1,233 @@
...
+    var birthdayBlockInnerHTML;
+    console.log('check');
+
+    for (var i = 1; i <= lastDate; i++) {
+        birthdaysOnDate = birthdays[String(_month + 1)][String(i)];
+        console.log(birthdaysOnDate);
+        birthdayBlockInnerHTML = '';
+        if (typeof(birthdaysOnDate) != "undefined") {
+            currentDateElement = document.querySelector('[data-month-date="' + (_month + 1) + '-' + i + '"]');
+            for (var j = 0; j < birthdaysOnDate.length; j++) {
```
Comment (shikhashakarwar) [inline Calendar/js/app.js]: you can use `birthdaysOnDate.join();` so no need of this loop.

File: Calendar/js/app.js [hunk 11]
```diff
@@ -0,0 +1,233 @@
...
+                birthdayBlockInnerHTML += '<p>' + birthdaysOnDate[j] + '</p>';
+                console.log(j);
+            }
+            currentDateElement.innerHTML = '<div id="bb" class="birthdayBlock">' + birthdayBlockInnerHTML + '</div>';
```
Comment (shikhashakarwar) [inline Calendar/js/app.js]: rename `id=bb`:-p

File: Calendar/js/app.js [hunk 12]
```diff
@@ -0,0 +1,233 @@
...
+            currentDateElement.innerHTML += i + ' <i class="fa fa-birthday-cake"></i>';
+            currentDateElement.style.color = "#F96120" //"#A5DE60"; //"#D01315"
```
Comment (shikhashakarwar) [inline Calendar/js/app.js]: make a class for this. as we discussed yesterday

File: Calendar/js/app.js (hunk 7 above)
Comment (MishraSagar) [reply Calendar/js/app.js]: Actually, I was using them to create calendar in single loop as anjana mam asked. I will remove them asap.

File: Calendar/js/app.js (hunk 12 above)
Comment (MishraSagar) [reply Calendar/js/app.js]: I changed the current date styling with a new class but forgot to do this. I will update the code asap.

**Pull request merged**